rpc/src/main/java/com/ibm/crail/rpc/RpcResponseMessage.java

```diff
@@ -135,15 +134,19 @@ public BlockInfo getFileBlock(){
 		}
 		
 		public void setFileBlock(BlockInfo blockInfo){
-			this.fileBlock.setBlockInfo(blockInfo);
+			if (blockInfo != null){
+				this.fileBlock.setBlockInfo(blockInfo);
+			}
 		}
 		
 		public BlockInfo getDirBlock(){
 			return dirBlock;
 		}
 		
 		public void setDirBlock(BlockInfo blockInfo){
-			this.dirBlock.setBlockInfo(blockInfo);
+			if (blockInfo != null){
```

Comment:
What happens here if the blockInfo is null? This code then silently drops it. IF we ever do not want to have blockInfo as null - then check that in else and throw an exception. 

In general - do not use null as a valid value that can be checked and used around. Because whenever we fail to take precautions, we get a null pointer exception and then we have to guess what went wrong.